module4/README.md

```diff
@@ -0,0 +1,50 @@
+Kodeco module 4 homework
+
+a) List two Collection Types in Swift.
+
+Array, Dictionary
+
+
+b) Name a function that uses variadic parameters.
+
+print()
+
+c) True or False: Only a class can adopt a protocol in Swift.
+
+False. Structs can adopt protocols, and protocols can inherit from other protocols
+
+d) Name the four Named Types you have been introduced to so far.
+
+Struct, Class, Enum, Protocol
+
+e) The variadic parameter is treated as an _____________ in the function.
+
+array
+
+f) In Swift, structs are ____________ types, and classes are _____________ types.
+
+value, reference
+
+g) True or False: Functions in Swift, by default, can alter parameters passed to the function.
+
+False. You must explicity use the inout keyword in front of the parameter name to be able to modify a parameter.
+
+h) ______________ are what a type has, and ____________ are what a type can do.
+
+properties, methods
+
+i) A class, struct, or enum that adopts a protocol is said to _______________ to the protocol
+
+conform
+
+j) A lazy property is calculated only when ________________________.
+
+it is accessed
```

Comment:
When it is _first_ accessed, not every time. A calculated property is every time.